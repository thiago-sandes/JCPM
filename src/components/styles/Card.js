import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  image: {width: 80, height: 120, marginRight: 10, borderRadius: 7},
  list: {
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 10,
    paddingBottom: 10,
    borderTopWidth: 1,
    borderBottomWidth: 1,
    borderLeftWidth: 1,
    borderRightWidth: 1,
    borderRadius: 10,
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 7,
    display: 'flex',
  },
  details: {
    fontSize: 18,
  },
  name: {
    width: '56%',
  },
});
