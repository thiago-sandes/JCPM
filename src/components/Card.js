import React from 'react';
import {View, Text, Image} from 'react-native';

// Styles
import styles from './styles/Card';

const Card = (props) => {
  const {title, key, year, img} = props;
  return (
    <View style={styles.list}>
      <View>
        <Image source={{uri: `${img}`}} style={styles.image} />
      </View>
      <View style={styles.name}>
        <Text style={styles.details} key={key}>
          {title}
        </Text>
        <Text>{year}</Text>
      </View>
    </View>
  );
};

export default Card;
