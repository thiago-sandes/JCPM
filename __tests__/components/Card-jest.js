import React from 'react';
import {TouchableOpacity} from 'react-native';
import renderer from 'react-test-renderer';

jest.useFakeTimers();

function fakeTimer(callback) {
  setTimeout(() => {
    callback && callback();
  }, 1000);
}

// Components
import Card from '../../src/components/Card';

// Mocks
const cardMock = {
  title: 'Avenger',
  id: 'tt0473445',
  year: '2006',
  img:
    'https://m.media-amazon.com/images/M/MV5BMTMzMjMwMjcyNl5BMl5BanBnXkFtZTcwNTA1NDgzMQ@@._V1_SX300.jpg',
  onClick: jest.fn(),
};

describe('Card component tests', () => {
  test('Card snapshot', () => {
    const component = renderer.create(
      <Card
        title={cardMock.title}
        id={cardMock.id}
        year={cardMock.year}
        img={cardMock.img}
        onClick={cardMock.onClick}
      />,
    );

    fakeTimer();

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Card onClick', () => {
    const component = renderer.create(
      <Card
        title={cardMock.title}
        id={cardMock.id}
        year={cardMock.year}
        img={cardMock.img}
        onClick={cardMock.onClick}
      />,
    );

    component.root.findByType(TouchableOpacity).props.onPress();
    expect(cardMock.onClick).toHaveBeenCalled();
  });
});
