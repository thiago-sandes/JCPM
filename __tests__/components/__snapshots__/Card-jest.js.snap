// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component tests Card snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRadius": 10,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 7,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://m.media-amazon.com/images/M/MV5BMTMzMjMwMjcyNl5BMl5BanBnXkFtZTcwNTA1NDgzMQ@@._V1_SX300.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 7,
            "height": 120,
            "marginRight": 10,
            "width": 80,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "56%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Avenger
      </Text>
      <Text>
        2006
      </Text>
    </View>
  </View>
</View>
`;
